### PR TITLE
Shortening the path of source files in source code nuget packages

### PR DESCRIPTION
**Pull request by popara96**
This PR will sort out long paths made by source code packages by unpacking all source files inside the packageDirectory/Sources.

### CODE REVIEW

File: src/NuGetForUnity/Editor/NugetPackageInstaller.cs [hunk 1]
```diff
@@ -196,36 +198,125 @@ private static bool Install([NotNull] INugetPackage package, bool refreshAssets,
...
+                                    "Selecting target framework directory 'any' and language 'any' as best match for the package {1}", package);
+
+                                PackageContentManager.ExtractPackageSources(anyFrameworks, baseDirectory);
+                            }
+                            else
+                            {
+                                Debug.LogWarningFormat(
+                                    "Couldn't find a source code folder with a supported target-framework for the package {0}",
+                                    package);
+                            }
```
Comment (igor84) [inline src/NuGetForUnity/Editor/NugetPackageInstaller.cs:319]: You can simplify this using something like:
```cs
if (csFiles.Count > 0)
{
	TryExtractBestFrameworkSources(csFiles, 'cs');
}
else if (anyFiles.Count > 0)
{
	TryExtractBestFrameworkSources(anyFiles, 'any');
}

private void TryExtractBestFrameworkSources([NotNull] [ItemNotNull] IReadOnlyCollection<KeyValuePair<string, List<ZipArchiveEntry>> frameworks, [NotNull] string sourceDirName)
{
	var bestFrameworkMatch = TargetFrameworkResolver.TryGetBestTargetFramework(frameworks, framework => framework.Key);
	var frameworkKey = bestFrameworkMatch.Key ?? "any";
	if (frameworks.TryGetValue(frameworkKey, out var bestFramework))
	{
		NugetLogger.LogVerbose(
			"Selecting target framework directory '{0}' and language '{1}' as best match for the package {2}",
			bestFramework,
			sourceDirName,
			package);

		PackageContentManager.ExtractPackageSources(bestFramework, baseDirectory);
	}
	else
	{
		Debug.LogWarningFormat(
			"Couldn't find a source code folder with a supported target-framework for the package {0}",
			package);
	}
}
```

File: src/NuGetForUnity/Editor/PackageContentManager.cs [hunk 2]
```diff
@@ -194,12 +203,47 @@ internal static bool ShouldSkipUnpackingOnPath([NotNull] string path)
...
+                {
+                    if (entryFullName[j] != lastCommonDir[j])
+                    {
+                        lastCommonDir = entryFullName.Substring(0, j);
+                        break;
+                    }
+                }
+            }
+
+            var sourcesDirectory = Path.Combine(baseDir, "Sources").Replace('\\', '/');
```
Comment (igor84) [inline src/NuGetForUnity/Editor/PackageContentManager.cs:228]: Why do you need this Replace here? I don't see you use this variable later in a way that requiers it.

**Review by igor84 — APPROVED** [review]

File: src/NuGetForUnity/Editor/PackageContentManager.cs (hunk 2 above)
Comment (popara96) [reply src/NuGetForUnity/Editor/PackageContentManager.cs:228]: It was useful when I was counting `/` occurrences, I will remove it now

**Review by JoC0de — CHANGES_REQUESTED** [review]
It would be nice to at least have one NuGet package that has source code to be installed by the Test so we can ensure this woks / doesn't brake later.
We could even create a ZipFile on-the-fly just for testing instead of using a existing .nupgk

File: src/NuGetForUnity/Editor/NugetPackageInstaller.cs (hunk 1 above)
Comment (JoC0de) [reply src/NuGetForUnity/Editor/NugetPackageInstaller.cs:319]: yes without the "extraction of a function" it is definitely to much copy-paste.

Comment (popara96) [reply src/NuGetForUnity/Editor/NugetPackageInstaller.cs]: Simplified but without `[ItemNotNull]` as Rider said `Applying nullness annotation to value type is meaningless`.

**Comment by popara96** [comment]
Will write tests tomorrow.

File: src/NuGetForUnity.Tests/Assets/Tests/Editor/NuGetTests.cs [hunk 3]
```diff
@@ -775,6 +775,37 @@ public void TestSlimRestoreInstall(string packageId, string packageVersion)
             package.Version);
     }
 
+    [Test]
+    [TestCase("Microsoft.Azure.WebJobs.Sources", "3.0.37")]
+    [TestCase("NServiceBus.Testing.Fakes.Sources", "7.1.13")]
+    public void TestSourceCodePackageInstall(string packageId, string packageVersion)
```
Comment (popara96) [inline src/NuGetForUnity.Tests/Assets/Tests/Editor/NuGetTests.cs:781]: I added a check to see if `Sources` directory is added in the package root and ensured that `contentFiles` directory doesn't exist. I'm trying to thing of other generic checks that we could do while installing this but nothing comes to mind as directories/files under `Sources` will be very package-specific. H

**Review by JoC0de — APPROVED** [review]
I didn't see any more issues, I would only fix the formatting at some places. See e.g. the Compiler warnings about trailing whitespace.

**Pull request merged**